libtock/alarm_timer.c

```diff
@@ -173,11 +175,16 @@ static void delay_upcall(__attribute__ ((unused)) int unused0,
   *((bool*)ud) = true;
 }
 
-void delay_ms(uint32_t ms) {
+int delay_ms(uint32_t ms) {
   bool cond = false;
   tock_timer_t timer;
-  timer_in(ms, delay_upcall, &cond, &timer);
+
+  if (timer_in(ms, delay_upcall, &cond, &timer) != RETURNCODE_SUCCESS) {
+    return RETURNCODE_FAIL;
```

Comment:
You should probably return the real result from `timer_in()` rather than `RETURNCODE_FAIL` here. That would make this code more future-proof. It's not really an error as-is though.

Reply:
Thanks, updated to return the actual return value of `timer_in()`.